ci_server/server.py

```diff
@@ -6,20 +6,22 @@ class CIServer(BaseHTTPRequestHandler):
     TODO: implement the CI server, this is just a placeholder.
     """
 
-    def do_GET(self):
+    def do_get(self):
```

Comment:
Is there a reason for changing this? I think it actually has to be `do_GET` and `do_POST` as those are the names that [BaseHTTPRequestHandler](https://docs.python.org/3/library/http.server.html#http.server.BaseHTTPRequestHandler) define for those methods, and since we are inheriting them I think we would have to name them the same. Have you tested making post/get requests with this changed naming convention?

Reply:
The reason that I changed it is due to the fact that it is not a valid snake_case function declaration according to PEP8 (https://pep8.org/) which is the basis of the default pylint linting rules. I will try to use these methods with the new names. If they don't work in their current format I will simply change linting rules to accommodate this.

Reply:
After testing I found that you were right. The methods had to comply with the original naming convention to work. Given that I don't see a reason to strictly enforce snake_case in this project I have decided to disable the C0103 (invalid-name) linting rule.